src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -205,11 +265,44 @@ private HttpStatusCode ParseStorageException(StorageException storageException)
                 }
                 catch (Exception)
```

Comment:
`Enum.TryParse` seems safer than catching a generic exception

Reply:
I thought of using `Enum.TryParse` initially, but then realized that the result would require further validation. As an example, `Enum.TryParse(intValue.ToString(), out HttpStatusCode responseCode)` with `intValue = 1` would return `True` and `responseCode = 1` even though it is an invalid HttpStatusCode. Hence I decided to use `IsDefined` since it would be easier and is the recommended solution in such situations (when converting int to string) - https://docs.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=netcore-3.1

Reply:
Ah got it, wasn't obvious that https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.requestresult.httpstatuscode?view=azure-dotnet#Microsoft_Azure_Storage_RequestResult_HttpStatusCode is an int :)
If this is the case, and since you are already using IsDefined, any reason we shouldn't just cast? `(HttpStatusCode)intValue`?

Reply:
The StorageException class does seem a bit weird :). Yes, I should be able to just cast. Will update.